src/views/blockDetails/blockHeader/index.tsx

```diff
@@ -22,22 +24,84 @@ interface DetailRowProps {
   label: string;
   value: string | undefined;
   isCopyable?: boolean;
+  iconComponent?: React.ReactNode;
 }
 
-const DetailRow = (props: DetailRowProps) => {
-  const { label, value } = props;
+export const DetailRow = (props: DetailRowProps) => {
+  const { label, value, iconComponent } = props;
+
+  const [
+    isHovered,
+    setIsHovered,
+  ] = useState(false);
+  const [
+    isParentHovered,
+    setIsParentHovered,
+  ] = useState(false);
+  const [
+    isMobile,
+    setIsMobile,
+  ] = useState<boolean>(window.innerWidth < 768);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobile(window.innerWidth < 768);
+    };
+
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+  const handleParentMouseEnter = useCallback(() => {
+    setIsParentHovered(true);
+  }, []);
+
+  const handleParentMouseLeave = useCallback(() => {
+    setIsParentHovered(false);
+  }, []);
+
+  const handleMouseEnter = useCallback(() => {
+    setIsHovered(true);
+  }, []);
+
+  const handleMouseLeave = useCallback(() => {
+    setIsHovered(false);
+  }, []);
```

Comment:
I think this could be achieved with just using CSS props

Reply:
Yes, you are right! I have refactored the implementation to achieve this using CSS props